src/common/load-plugins.js

```diff
@@ -25,11 +26,48 @@ function load(plugins, pluginSearchDirs) {
   if (!pluginSearchDirs) {
     pluginSearchDirs = [];
   }
-  // unless pluginSearchDirs are provided, auto-load plugins from node_modules that are parent to Prettier
+  // unless pluginSearchDirs are provided, auto-load plugins from
+  // the first node_modules folder that is parent to the cwd
+
+  const findPluginDirsInPnpmVirtualStore = (searchDirectory) => {
+    const pluginPackageJsonPaths = fastGlob.sync(
+      ["prettier-plugin-*", "@*/node_modules"],
+      {
+        cwd: searchDirectory,
+        onlyDirectories: true,
+      }
+    );
+    return pluginPackageJsonPaths.map((match) =>
+      path.resolve(searchDirectory, match)
+    );
+  };
+
   if (pluginSearchDirs.length === 0) {
-    const autoLoadDir = thirdParty.findParentDir(__dirname, "node_modules");
-    if (autoLoadDir) {
-      pluginSearchDirs = [autoLoadDir];
+    // This is likely the project root. In projects using pnpm, this directory is named "prettier@<version>".
+    let autoLoadDirectory = thirdParty.findParentDir(__dirname, "node_modules");
+    autoLoadDirectory = autoLoadDirectory
+      ? path.resolve(autoLoadDirectory)
+      : ""; // Make sure autoLoadDir isn't null.
+
+    const isProjectRoot = findProjectRoot(__dirname) === autoLoadDirectory; // Make sure we don't go outside projectRoot.
```

Comment:
Though it's not recommend, but we support global installation, I think editors use global one when Prettier is not installed locally.

Reply:
That shouldn't affect global install. `isProjectRoot` is only used to check if we're in the pnpm virtual store (line 57).